include/lsst/meas/base/PixelFlags.h

```diff
@@ -94,10 +110,12 @@ class PixelFlagsAlgorithm : public SimpleAlgorithm {
     ) const;
 
 private:
-
+    template <typename MaskedImageT>
+    class FootprintBits;
+    typedef afw::image::MaskedImage<float> MaskedImageT;
+    void updateFlags(const KeyMap *,const FootprintBits<MaskedImageT> *,afw::table::SourceRecord & ) const;
```

Comment:
Hmm, I'm not sure if I agree with the general point, but I might if you shouted loud enough.

In this case, though, const references are fine -- only the `SourceRecord` (which is already a non-const reference) is being modified.